docs/features/async-streams.md

```diff
@@ -182,13 +183,30 @@ IAsyncEnumerator<T> GetAsyncEnumerator(CancellationToken token)
     {
         result = new {StateMachineType}(InitialState);
     }
-    /* copy each parameter proxy, or copy the token parameter if it's not default for any parameter marked with [EnumeratorCancellation] attribute */
+    /* copy each parameter proxy, or in the case of the parameter marked with [EnumeratorCancellation] combine it with `GetAsyncEnumerator`'s `token` parameter */
+}
+```
+
+For the parameter with `[EnumeratorCancellation]`, `GetAsyncEnumerator` initializes it by combining the two available tokens:
+```csharp
+if (this.parameterProxy.Equals(default))
+{
+    result.parameter = token;
+}
+else if (token.Equals(this.parameterProxy) || token.Equals(default))
+{
+    result.parameter = this.parameterProxy;
+}
+else
+{
+    result.combinedTokens = CancellationTokenSource.CreateLinkedTokenSource(this.parameterProxy, token);
```

Comment:
>combinedTokens [](start = 11, length = 14)

Should I be seeing this disposed in the description of the translated code below?

Reply:
Because the `MoveNext` method is rather large/complicated, I did not show the code in this document. I just mentioned that we dispose combinedFields (above, line 137), without providing a code snippet.

---
In reply to: [279965890](https://github.com/dotnet/roslyn/pull/35326#discussion_r279965890) [](ancestors = 279965890)